payments-core/src/test/java/com/stripe/android/googlepaylauncher/utils/GooglePayLauncherTestRunner.kt

```diff
@@ -0,0 +1,124 @@
+) {
+    for (integrationType in integrationTypes) {
+        runGooglePayLauncherTest(
+            integrationType = integrationType,
+            isReady = isReady,
+            result = result,
+            verifyResult = expectResult,
+            block = block,
+        )
+    }
+}
+
+private fun runGooglePayLauncherTest(
+    integrationType: LauncherIntegrationType,
+    isReady: Boolean,
+    result: GooglePayLauncher.Result,
+    verifyResult: Boolean,
+    block: (ComponentActivity, GooglePayLauncher) -> Unit,
+) {
+    val readyCallback = mock<GooglePayLauncher.ReadyCallback>()
+    val resultCallback = mock<GooglePayLauncher.ResultCallback>()
+
+    Mockito.mockStatic(Wallet::class.java).use { wallet ->
+        val paymentsClient = mock<PaymentsClient>()
+        whenever(paymentsClient.isReadyToPay(any())).thenReturn(Tasks.forResult(isReady))
+
+        // Provide a static mock here because DefaultGooglePayRepository calls this static
+        // create method.
+        wallet.`when`<PaymentsClient> {
+            Wallet.getPaymentsClient(isA<Context>(), any())
+        }.thenReturn(paymentsClient)
+
+        // Mock the return value from GooglePayLauncherActivity so that we immediately return
+        val resultData = Intent().putExtras(
+            bundleOf(GooglePayLauncherContract.EXTRA_RESULT to result)
+        )
+        val activityResult = Instrumentation.ActivityResult(Activity.RESULT_OK, resultData)
+        intending(anyIntent()).respondWith(activityResult)
+
+        val scenario = ActivityScenario.launch(ComponentActivity::class.java)
+        scenario.moveToState(Lifecycle.State.CREATED)
+
+        scenario.onActivity { activity ->
+            PaymentConfiguration.init(activity, "pk_test")
+
+            lateinit var launcher: GooglePayLauncher
+
+            when (integrationType) {
+                LauncherIntegrationType.Activity -> {
+                    launcher = GooglePayLauncher(
+                        activity = activity,
+                        config = defaultConfig,
+                        readyCallback = readyCallback,
+                        resultCallback = resultCallback,
+                    )
+                }
+                LauncherIntegrationType.Compose -> {
+                    activity.setContent {
+                        launcher = GooglePayLauncher.rememberLauncher(
+                            config = defaultConfig,
+                            readyCallback = readyCallback,
+                            resultCallback = resultCallback,
+                        )
+                    }
+                }
+            }
+
+            scenario.moveToState(Lifecycle.State.RESUMED)
+            block(activity, launcher)
+        }
+
+        Espresso.onIdle()
+    }
+
+    verify(readyCallback).onReady(eq(isReady))
+
+    if (verifyResult) {
+        verify(resultCallback).onResult(eq(result))
+    }
+
+    reset(readyCallback, resultCallback)
```

Comment:
Why is this needed?

Reply:
Oh, I think this is no longer necessary.